### PR TITLE
Enable build pipeline for non-code-signed Nuget, refactor Nuget pipeline, and clean-up

### PR DESCRIPTION
**Pull request by jefgen**
## Summary
This PR refactors the Nuget build pipeline for the MS-ICU Nuget package.

This changes the Nuget build pipeline to enable code-signing, and also allows for running the pipeline in two different modes:
- Code signing enabled.
- No code signing enabled.

This allows for running the Nuget pipeline as a "CI check" on PRs, in order to ensure that any changes don't accidentally break things for the Nuget package. With this change we can now switch over to use this new YAML-based Nuget build pipeline configuration instead of the legacy "classic" ADO build pipeline configuration.
Furthermore, since the Nuget pipeline is now checked-in to the repo, it means that anyone can edit the configuration (ie: to add a new platform). Whereas before it was only in MSCodeHub, and was not accessible at all.

Also, the guidance from discussion with the PackageES team is that the `ComponentGovernance` task must be *after* all of the various build task(s) are done (that produces binaries).

This PR also makes a few other changes as well:
- The contents of the build artifacts that are published across stages is reduced.
- PackageES tasks are all skipped in the Non-code sign mode, in order to speed things up.
- The non-signed Nuget package is published as a build artifact in order to allow testing/inspection of changes.
- The non-signed Nuget package uses `-prerelease` as a suffix along with the monotonic increasing build number.
- Removes the two PowerShell scripts that were used to set variables in the build pipeline, as we can do this directly in the YAML file now.

## PR Checklist
* [x] I have verified that my change is specific to this fork and cannot be made upstream.
* [ ] I am making a maintenance related change.
* [x] I am making a change that is related to usage internal to Microsoft.
* [ ] I am making a change that is related to the Windows OS build of ICU.
* [x] CLA signed. If not, please see [here](https://cla.opensource.microsoft.com/microsoft/icu) to sign the CLA.

### CODE REVIEW

File: build/azure-nuget.yml [hunk 1]
```diff
@@ -123,7 +150,10 @@ stages:
       fetchDepth: 1
     - task: PkgESSetupBuild@10
       displayName: 'PkgES Setup Build'
+      env:
+          SYSTEM_ACCESSTOKEN: $(system.accesstoken)
```
Comment (daniel-ju) [inline build/azure-nuget.yml:154]: Do we need this on tasks that aren't code-signing?

Comment (jefgen) [reply build/azure-nuget.yml:154]: Fair question. :)
I honestly don't think we technically need to do it, as when I was testing it was working without it. However, I wanted to replicate the existing pipeline as close as possible. The steps that run on the PackageES agents all have the token set at the run level, meaning that all tasks/jobs have access to it (even if they don't need it). So this is really just me wanting to be as close as possible to the existing setup.

Comment (daniel-ju) [reply build/azure-nuget.yml:154]: Hmm that's fine with me if we're trying to align with the existing setup, but maybe we can remove it from both pipelines in the future if it's not needed?

Comment (jefgen) [reply build/azure-nuget.yml]: I can't find any docs that say it's explicitly required, so I'll remove it for now.

Comment (daniel-ju) [reply build/azure-nuget.yml]: Thanks Jeff! :)

**Pull request merged**